test/framework/cluster/validations/cluster.go

```diff
@@ -60,6 +60,35 @@ func ValidateEKSAObjects(ctx context.Context, vc clusterf.StateValidationConfig)
 	return nil
 }
 
+// ValidateBundleKubernetesTag is a loose check for checking  that the bundle is in use by the cluster.
```

Comment:
for example, it would be good to check if the correct version of the CNI is installed and other key components.

I'm not really convinced the right solution for this yet is to validate the state of the cluster to the bundle because it is someone complex to do so. Therefore, the task was made to also explore exactly what needs to be done here as well.